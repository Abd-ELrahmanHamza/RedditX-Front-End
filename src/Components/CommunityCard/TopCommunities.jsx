import { CommunityContainer } from "./TopCommunities.styled";
import { Link } from "react-router-dom";
import CommunityCardItem from "../CommunityCardItemExplore/CommunityCardItemExplore";
import axios from "API/axios";
import useFetch from "Hooks/useFetch";
/**
 * Component that contains the Community Card Elements.
 *
 * @Component
 * @returns {React.Component}
 */
const TopCommunities = ({ CommunityList, communityListSub }) => {
  // let [CommunityList, error, loading, reload] = useFetch({
  //   axiosInstance: axios,
  //   method: "GET",
  //   url: "http://localhost:8000/communityList",
  //   requestConfig: {
  //     headers: {
  //       "Content-Language": "en-US",
  //     },
  //   },
  // });
  if (CommunityList.results && communityListSub) {
    const arr = CommunityList.results;
    console.log(arr);
    const communitiesNumber = CommunityList.results.length;
    return (
      <CommunityContainer>
        <p className="title">Top Communities</p>

        {communitiesNumber !== 0 &&
          CommunityList &&
          CommunityList.results &&
          communityListSub &&
          communityListSub.communities &&
          CommunityList.results.slice(0, 8).map((Community) => (
            <CommunityCardItem
              communityIcon={Community.icon}
              key={Community._id}
              communityID={Community._id}
              communityDescription={Community.description}
              membersCount={Community.membersCnt}
              communityName={Community._id}
              isJoined={communityListSub.communities.find((element) => {
                return element._id === Community._id;
              })}
            />
          ))}
        {communitiesNumber === 0 && <p className="NoResults">No results</p>}
        {communitiesNumber !== 0 && (
          <Link to="/explore/Communities">
            <p className="more">See more</p>
          </Link>
        )}
      </CommunityContainer>
    );
  }
};

export default TopCommunities;
