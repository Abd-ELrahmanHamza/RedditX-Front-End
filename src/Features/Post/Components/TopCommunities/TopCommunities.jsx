import { CommunityContainer } from "./TopCommunities.styled";
import CommunityCardItem from "../CommunityCardItem/CommunityCardItem";
import community from "../../../Subreddit/Components/Community/Community";

const CommunityList = [
  {
    communityID: "1",
    communityDescription: "we are looking for a new community",
    membersCount: "4",
    communityName: "Cufe",
  },
  {
    communityID: "2",
    communityDescription: "we are looking for a new modal",
    membersCount: "1",
    communityName: "ASU",
  },
  {
    communityID: "3",
    communityDescription: "we are looking for a new school",
    membersCount: "1",
    communityName: "MCE",
  },
];
/**
 * Component that contains the community card elements (container for community card elements).
 *
 * @Component
 * @param {object} CommunityList -  contains the Community list elements
 * @param {object} CommunitiesSub2 -  contains the Community list elements that user subscribe
 * @returns {React.Component}
 */
const TopCommunities = () => {
  return (
    <CommunityContainer>
      <p className="title">Communities</p>
      {CommunityList.map((Community) => (
        <CommunityCardItem
          // communityIcon={Community.communityIcon}
          key={Community.communityID}
          communityID={Community.communityID}
          communityDescription={Community.communityDescription}
          membersCount={Community.membersCount}
          communityName={Community.communityName}
        />
      ))}
    </CommunityContainer>
  );
};

export default TopCommunities;
