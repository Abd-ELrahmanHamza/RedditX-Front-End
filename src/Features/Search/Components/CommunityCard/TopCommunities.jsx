import { CommunityContainer } from "./TopCommunities.styled";
// import { useState } from "react";
// import logo from "../../Assets/download.jpg";
import { Link } from "react-router-dom";
import CommunityCardItem from "../CommunityCardItem/CommunityCardItem";
const TopCommunities = () => {
  return (
    <CommunityContainer>
      <p className="title">Communities</p>
      <CommunityCardItem />
      <CommunityCardItem />
      <CommunityCardItem />
      <CommunityCardItem />
      <CommunityCardItem />
      <Link to="/communities">
        <p className="more">See more communities</p>
      </Link>
    </CommunityContainer>
  );
};

export default TopCommunities;
