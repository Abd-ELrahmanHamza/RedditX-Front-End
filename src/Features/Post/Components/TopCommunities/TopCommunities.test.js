import { shallow } from "enzyme";

import TopCommunities from "./TopCommunities";

describe("Button Component", () => {
  it("this is a test for Button Component", () => {
    expect(shallow(<TopCommunities />)).toMatchSnapshot();
  });
});
