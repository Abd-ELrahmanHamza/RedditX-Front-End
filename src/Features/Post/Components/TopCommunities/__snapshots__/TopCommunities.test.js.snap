// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component this is a test for Button Component 1`] = `
<styled.div>
  <p
    className="title"
  >
    Communities
  </p>
  <CommunityCardItem
    communityDescription="we are looking for a new community"
    communityID="1"
    communityName="Cufe"
    key="1"
    membersCount="4"
  />
  <CommunityCardItem
    communityDescription="we are looking for a new modal"
    communityID="2"
    communityName="ASU"
    key="2"
    membersCount="1"
  />
  <CommunityCardItem
    communityDescription="we are looking for a new school"
    communityID="3"
    communityName="MCE"
    key="3"
    membersCount="1"
  />
</styled.div>
`;
